src/lightning/app/cli/commands/cp.py

```diff
@@ -0,0 +1,221 @@
+from pathlib import Path
+from time import sleep
+from typing import Optional, Tuple
+
+import click
+import requests
+import rich
+import urllib3
+from lightning_cloud.openapi import IdArtifactsBody
+from rich.live import Live
+from rich.progress import BarColumn, DownloadColumn, Progress, Task, TextColumn
+from rich.spinner import Spinner
+from rich.text import Text
+
+from lightning.app.cli.commands.pwd import _pwd
+from lightning.app.source_code import FileUploader
+from lightning.app.utilities.app_helpers import Logger
+from lightning.app.utilities.network import LightningClient
+
+logger = Logger(__name__)
+
+
+@click.argument("src_path", required=True)
+@click.argument("dst_path", required=True)
+@click.option("-r", required=False, hidden=True)
+@click.option("--recursive", required=False, hidden=True)
+def cp(src_path: str, dst_path: str, r: bool = False, recursive: bool = False) -> None:
+    """Command to copy files between your local and the Lightning Cloud filesystem's."""
+
+    with Live(Spinner("point", text=Text("pending...", style="white")), transient=True) as live:
+
+        pwd = _pwd()
+
+        if pwd == "/" or len(pwd.split("/")) == 1:
+            return _error_and_exit("Uploading files at the project level isn't supported yet.")
+
+        client = LightningClient()
+
+        src_path, src_remote = _sanetize_path(src_path, pwd)
+        dst_path, dst_remote = _sanetize_path(dst_path, pwd)
+
+        if src_remote and dst_remote:
+            return _error_and_exit("Moving files remotely isn't support yet. Please, open a Github issue.")
+
+        if not src_remote and dst_remote:
+            _upload_files(live, client, src_path, dst_path, pwd)
+        elif src_remote and not dst_remote:
+            _download_files(live, client, src_path, dst_path, pwd)
+        else:
+            return _error_and_exit("Moving files locally isn't support yet. Please, open a Github issue.")
+
+
+def _upload_files(live, client: LightningClient, local_src: str, remote_dst: str, pwd: str) -> str:
+    if not os.path.exists(local_src):
+        return _error_and_exit(f"The provided source path {local_src} doesn't exist.")
+
+    project_id, app_id = _get_project_app_ids(pwd)
+
+    local_src = Path(local_src).resolve()
+    upload_paths = []
+
+    if os.path.isdir(local_src):
+        for root_dir, _, paths in os.walk(local_src):
+            for path in paths:
+                upload_paths.append(os.path.join(root_dir, path))
+    else:
+        upload_paths = [local_src]
+
+    upload_urls = []
+
+    for upload_path in upload_paths:
+        filename = str(upload_path).replace(str(os.getcwd()), "")[1:]
+        response = client.lightningapp_instance_service_upload_lightningapp_instance_artifact(
+            project_id=project_id,
+            id=app_id,
+            body=IdArtifactsBody(filename),
+            async_req=True,
+        )
+        upload_urls.append(response)
+
+    live.stop()
+
+    sleep(1)
```

Comment:
Maybe a comment here why it has to sleep

Reply:
Done !